packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1047,11 +1057,7 @@ class TextInput {
     // the client ID, as long as we have a _currentConnection.
     if (method == 'TextInputClient.requestExistingInputState') {
       assert(_currentConnection._client != null);
-      _attach(_currentConnection, _currentConfiguration);
-      // This will be null if we've never had a call to [_setEditingState].
-      if (_currentTextEditingValue != null) {
```

Comment:
I wonder if the code would be a little bit clearer leaving this if outside the _attach method. The setEditingState seems like kinda a strange parameter to that method.

Reply:
Sure - it seemed a little awkward to me either way but this way it'll be closer to the actual use case for it.

Reply:
Actually redoing this uncovered a potential bug (TextEditingValue  can be null).